src/components/BackgroundMedia/__tests__/BackgroundMedia.test.tsx

```diff
@@ -0,0 +1,151 @@
+
+import {testAnimated, testCustomClassName} from '../../../../test-utils/shared/common';
+import {qaIdByDefault as qaIdOfBackgroundImage} from '../../../components/BackgroundImage/BackgroundImage';
+import {defaultAnimatedDivQa} from '../../../components/Media/Image/Image';
+import {qaIdByDefault as qaIdOfMedia} from '../../../components/Media/Media';
+import {
+    defaultMediaVideoQa,
+    defaultMediaVideoSourceQa,
+} from '../../../components/Media/Video/Video';
+import {BackgroundMediaProps} from '../../../models';
+import BackgroundMedia from '../BackgroundMedia';
+
+const qaId = 'background-media-component';
+
+const imageUrl =
+    'https://storage.yandexcloud.net/cloud-www-assets/constructor/storybook/images/img-gray.png';
+const videoProps = {
+    src: [
+        'https://storage.yandexcloud.net/cloud-www-assets/constructor/storybook/images/header-bg-video_light.mp4',
+    ],
+};
+
+describe('BackgroundMedia', () => {
+    test('render component by default', async () => {
+        render(<BackgroundMedia qa={qaId} />);
+        const component = screen.getByTestId(qaId);
+
+        expect(component).toBeInTheDocument();
+    });
+
+    test('add className', () => {
+        testCustomClassName<BackgroundMediaProps>({
+            component: BackgroundMedia,
+            props: {qa: qaId},
+        });
+    });
+
+    test('render with animated', async () => {
+        testAnimated<BackgroundMediaProps>({
+            component: BackgroundMedia,
+            props: {qa: qaId},
+        });
+    });
+
+    test('render with passed color', () => {
+        const style = {backgroundColor: 'red'};
+
+        render(<BackgroundMedia color={style.backgroundColor} qa={qaId} />);
+        const component = screen.getByTestId(qaId);
+
+        expect(component).toHaveStyle(style);
+    });
+
+    test('add className to media component', () => {
+        const className = 'my-class';
+
+        render(<BackgroundMedia qa={qaId} mediaClassName={className} />);
+        const component = screen.getByTestId(qaIdOfMedia);
+
+        expect(component).toHaveClass(className);
+    });
+
+    test('render video', () => {
+        render(<BackgroundMedia qa={qaId} video={videoProps} />);
+        const component = screen.getByTestId(defaultMediaVideoSourceQa);
+
+        expect(component).toBeInTheDocument();
+    });
+
+    test('render video with videoClassName', () => {
+        render(<BackgroundMedia qa={qaId} video={videoProps} />);
+        const component = screen.getByTestId(defaultMediaVideoQa);
+
+        expect(component).toHaveClass('pc-BackgroundMedia__video');
+    });
+
+    test('render video with default height', () => {
+        const style = {height: '720px'};
+        render(<BackgroundMedia qa={qaId} video={videoProps} />);
+        const component = screen.getByTestId(defaultMediaVideoQa);
+
+        expect(component).toHaveStyle(style);
+    });
+
+    test('render video with defined height', () => {
```

Comment:
maybe custom instead of defined?
Here and in other places

Reply:
sure